azurelinuxagent/ga/cgroupcontroller.py

```diff
@@ -40,12 +39,12 @@ class MetricValue(object):
     Class for defining all the required metric fields to send telemetry.
     """
 
-    def __init__(self, category, counter, instance, value, report_period=_DEFAULT_REPORT_PERIOD):
+    def __init__(self, category, counter, instance, value, report_period=None):
         self._category = category
         self._counter = counter
         self._instance = instance
         self._value = value
-        self._report_period = report_period
+        self._report_period = timedelta(seconds=conf.get_cgroup_check_period()) if report_period is None else report_period
```

Comment:
why make this change? Doesn't keeping report_period=_DEFAULT_REPORT_PERIOD achieve the same thing?

Reply:
Actually, this is actually a bug. This _DEFAULT_REPORT_PERIOD global loads during module imports, before config loaded from the waagent.conf. As a result, _DEFAULT_REPORT_PERIOD gets a default value even config is present in the file. We addressed similar issue in other places where we reference config in the global imports, I missed here somehow.

Reply:
Would changing the default value of report_period from the global to `conf.get_cgroup_check_period()` fix the issue?

 `def __init__(self, category, counter, instance, value, report_period=conf.get_cgroup_check_period()):`